### PR TITLE
omero.server build from source for testing integration

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
This helps running integration tests against local server

cc: @manics @sbesson @joshmoore @mtbc @jburel

### CODE REVIEW

File: omero-dev-runtests/buildomero.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+set +o nounset
+source ~omero/omero-virtualenv/bin/activate
+set -o nounset
+
+pip install --upgrade pip
+pip install --upgrade 'Pillow<3.0'
+pip install --upgrade omego
+
+
+## download source
```
Comment (aleksandra-tarkowska) [inline omero-dev-runtests/buildomero.sh]: If that is too complicated to maintain I can add ZIPURL the same like in https://github.com/ome/ome-docker/blob/master/omero-web/omeroweb.sh#L16

File: omero-dev-runtests/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+RUN yum install -y ice ice-python-devel ice-java-devel ice-servers && \
+    yum clean all
+
+RUN yum install -y git \
+    zlib-devel \
+    libjpeg-devel \
+    gcc && \
+    yum clean all
+
+RUN yum install -y python-pillow \
```
Comment (manics) [inline omero-dev-runtests/Dockerfile]: You're missing an `&& \` at the end

File: omero-dev-runtests/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    yum clean all
+
+RUN yum -y install \
+    http://yum.postgresql.org/9.4/redhat/rhel-7-x86_64/pgdg-centos94-9.4-2.noarch.rpm \
```
Comment (manics) [inline omero-dev-runtests/Dockerfile]: Also missing `&&`. If you want you could merge the next line in too e.g.

```
RUN yum -y install \
    http://yum.postgresql.org/9.4/redhat/rhel-7-x86_64/pgdg-centos94-9.4-2.noarch.rpm && \
    yum -y install postgresql94-server postgresql94 && \
    yum clean all
```

and similarly for some of the other blocks, but that's up to you.

**Comment by joshmoore** [comment]
This is great, @aleksandra-tarkowska. Beyond @manics comment(s), moving forward I think there's some potential for refactoring this to use https://github.com/ome/ome-docker/tree/master/omero-grid (or similar) which should help reduce the maintenance overhead. When it's time for that, let's move this to it's own repo and hook it up to hub.docker.com so there are official builds that everyone can use.

**Comment by jburel** [comment]
Note that on omero-install, we install pillow using pip so we have the desired version (2.9 soon 3.1)

**Comment by aleksandra-tarkowska** [comment]
@jburel I created the card https://trello.com/c/qKgqMiNj/640-which-prerequisite-we-should-support about what exactly should be tested. I think we should discuss that sooner then later.

If someone prefer newer version can uncomment https://github.com/ome/ome-docker/pull/31/files#diff-78fd7bdf3fe005a20ec8f87888fdf9e6R16

**Comment by aleksandra-tarkowska** [comment]
@manics yum installed dependencies updated

File: omero-dev-runtests/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+1. Alter omero config in settings.env as needed.
+
+2. Add tests to runtest.sh
+
+3. Build:
+
+        docker build -rm -t omerotest .
+
+    with custom zip:
```
Comment (manics) [inline omero-dev-runtests/README.md]: Is this old text that shouold be deleted?

**Comment by manics** [comment]
Looks good! A few suggested changes: https://github.com/manics/ome-docker/commit/48af96c77b17b91f36a1b89614d645bcebdeaad5 (take it or leave it). There's a truncated sentence in the README. `/home/omero/omero-source/build.py -f components/tools/OmeroWeb/build.xml integration` doesn't run due to lack of django, assuming this is intentional could you either add a comment to `runtests.sh` or in the README?

Apart from these points I think this is good to merge.

**Comment by aleksandra-tarkowska** [comment]
@manics your commit is here now. Django wasn't intentional. I also moved pip requirements to the separate file. I hope it makes more sense now.

**Comment by joshmoore** [comment]
Merging in order to investigate refactoring with omero-install usage. At that point, we likely want to migrate this to its own repo in order to have it built regularly by hub.docker.com

Note: tests in runtest.sh are still commented out

**Pull request merged**